### DISCUSSION TITLE
How to convert the code compiled by yalmip to c.

### DISCUSSION THREAD

**Discussion by liangjunfu** [yalmip/YALMIP#1270] [General discussion]
Dear Johan, I meet a new problem.I would like to ask how to convert the following code (showing model predictive control) to c.

clear;
clc;

% Model data
A = [2 -1;1 0.2];
B = [1;0];
nx = 2; % Number of states
nu = 1; % Number of inputs

% MPC data
Q = eye(2);
R = 2;
N = 7;
u = sdpvar(repmat(nu,1,N),repmat(1,1,N));
x= sdpvar(repmat(nx,1,N+1),repmat(1,1,N+1));
raodong=sdpvar(2,1);

constraints = [];
objective = 0;

for k = 1:N
objective = objective + x{k}'*Q*x{k} + u{k}'*R*u{k};
% constraints = [constraints, x{k+1} == Ax{k} + B*u{k1}；
constraints = [constraints,x{k+1} == A*x{k} + B*u{k}+raodong];
constraints = [constraints, -1 <= u{k}<= 1, -5<=x{k+1}<=5];
end
constraints=[constraints ,0.01<=raodong<=1]
[sol,diagnostics,aux,Valuefunction,Optimalsolution]=solvemp(constraints,objective,[],raodong);

assign(x{1},[-1;1]);
value(Valuefunction)    
value(Optimalsolution)   
figure;plot(Valuefunction)
figure;plot(Optimalsolution)

**Comment by johanlofberg** [comment]
That sounds like an impossible to task (well, you could always try to run mcc or what ever the compiler is called these days but I don't think it will be able to compile it to c(

Since you are computing a multiparametric solution, don't you want to convert the multiparametric solution with associated look-up to C? If so, you will have to study the MPT manuals, and then use the solution you have computed and get that to work with the stuff available in MPT to derive code.

**Reply by liangjunfu** [reply]
Thank you for your advice. I have already understood how to convert the multi-parameter solution of empc and the related lookup to C in MPT3, but I still can't understand how to convert yalmip to c in the reference to the MPT3 user manual, that is, how to convert the multi-parameter solution and the related lookup to C.

    Do you have any suggestions or examples for converting multiparameter solutions and associated look-up to C.

    I really appreciate your patient reply

**Reply by johanlofberg** [reply]
As I said, it makes no sense to convert the YALMIP code to C. The only thing that makes sense is to take the  MPT data "sol" and use that in some of MPTs code generation function (although you might have to massage the structure as it is MPT2 format and I have no idea if that is supported in the code generation parts of MPT)
